lib/functions.php

```diff
@@ -208,6 +208,15 @@ function ($matches) {
     if (!$result) {
         $result = _parse_fallback($uri);
     }
+    else
+    {
+        // Add empty host and trailing slash to windows file paths (file:///C:/path)
+        if (isset($result['scheme']) && $result['scheme'] === 'file' && isset($result['path']) &&
+             preg_match('/^(?<windows_path> [a-zA-Z]:(\/(?![\/])|\\\\)[^?]*)$/x', $result['path'])) {
```

Comment:
The regex also matches something like `C:\\abc\def.txt` and `C:\\abc/def.txt` - as well as `C:/abc/def.txt`

In what situations does `C:\\` happen?

@peterpostmann @staabm @evert - I have rebased this in #71 and want to make sure I understand all the combinations and why they need to be handled. (And I will write some comment lines in the code in that PR, to help future devs who have forgotten all the detail of Windows paths)

Reply:
I don't really have the context anymore to provide useful input im afraid =(

Reply:
https://docs.microsoft.com/en-us/dotnet/standard/io/file-path-formats is a place where I can find `C:\\` ;
`Console.WriteLine("Setting current directory to 'C:\\'");`

But that is C# code, and the `\\` is just to put a literal `\` in the string. The actual string is:
`Setting current directory to 'C:\'`

So that does not give me an example of how code can get here and have `C:\\` in the string, and be valid representing something in the MS world.

Reply:
Parsing ```file:///C:\\abc\\def.txt``` is something you probably want to do to process path strings with escaped backslashes. But revisiting, I would simplify the logic and remove the restrictions. I think the idea was to only process ```C:\``` (valid path),  ```C:\\``` (valid escaped path),  ```C:/``` (valid path with forward slashes), but not ```C://``` because this is not a valid Windows path anyways. But since the fallback is parsing ```C://``` nevertheless, we should do the same to avoid inconstancy.

```php
// Add empty host and trailing slash to windows file paths (file:///C:/path)
if (isset($result['scheme']) && $result['scheme'] === 'file' && isset($result['path']) &&
     preg_match('/^(?<windows_path> [a-zA-Z]:(\/|\\\\).*)$/x', $result['path'])) {
    $result['path'] = '/' . $result['path'];
    $result['host'] = '';
}
```

This will just look for a single letter followed by a colon and a slash, which is consistent with the fallback

```
=== file:///C:\abc\def.txt:
old parse: scheme: file, path: C:\abc\def.txt
fallback:  scheme: file, path: /C:\abc\def.txt
new parse: scheme: file, path: /C:\abc\def.txt

=== file:///C:\abc/def.txt:
old parse: scheme: file, path: C:\abc/def.txt
fallback:  scheme: file, path: /C:\abc/def.txt
new parse: scheme: file, path: /C:\abc/def.txt

=== file:///C:/abc/def.txt:
old parse: scheme: file, path: C:/abc/def.txt
fallback:  scheme: file, path: /C:/abc/def.txt
new parse: scheme: file, path: /C:/abc/def.txt

=== file:///C://abc/def.txt:
old parse: scheme: file, path: C://abc/def.txt
fallback:  scheme: file, path: /C://abc/def.txt
new parse: scheme: file, path: /C://abc/def.txt
```

Reply:
@peterpostmann I implemented this kind of thing now in #71 - that seems to work fine.